commands/application.angular.js

```diff
@@ -17,34 +20,81 @@ function runSchematicCommand(schematicCommand, options, evaluatingOptions) {
         additionalOptions.push(schematicOption);
     };
 
-    const commandArguments = [
-        '-p', '@angular/cli@latest',
-        '-p', collectionPath,
+
+    let commandArguments = [
         'ng', 'g', `${collectionName}:${schematicCommand}`
     ].concat(additionalOptions);
 
-    runNpxCommand(commandArguments, evaluatingOptions);
+    const packageName = getNotInstalledLocalPackage(collectionName, evaluatingOptions && evaluatingOptions.cwd);
+
+    checkAngularPackage().then((result) => {
+        if(result) {
+            commandArguments = result.concat(commandArguments);
+        }
+
+        if(packageName) {
+            runCommand('npm', ['install', collectionName]).then(() => {
+                runCommand('npx', commandArguments, evaluatingOptions);
+            });
+        } else {
+            runCommand('npx', commandArguments, evaluatingOptions);
+        }
+    });
+}
+
+function getNotInstalledLocalPackage(packageName, cwd) {
+    if(fs.existsSync('node_modules') || cwd) {
+        const packageJsonPath = path.join(cwd || process.cwd(), `node_modules/${packageName}/package.json`);
+
+        if(!fs.existsSync(packageJsonPath)) {
+            return packageName;
+        }
+    }
+    return;
+}
+
+function checkAngularPackage() {
+    const minVersion = new semver('6.0.0');
+    return new Promise((resolve, reject) => {
+        exec('ng v', (err, stdout, stderr) => {
+            const commandArguments = ['-p', '@angular/cli'];
+            if(stderr) {
+                resolve(commandArguments);
+            }
+            if(stdout) {
+                const commandResult = stdout.toString();
+                const version = new semver(commandResult.match(/(?<=angular.cli:.)([0-9]+.[0-9]+.[0-9]+)/ig)[0]);
```

Comment:
`angular.cli` could be replaced with `@angular\/cli`
`([0-9]+.[0-9]+.[0-9]+)` does not consider unstable versions with `-beta.*` tail